docs/backend-system/building-backends/08-migrating.md

```diff
@@ -1063,3 +1063,289 @@ backend.start();
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-pg/alpha'));
+/* highlight-add-end */
+```
+
+##### Elasticsearch
+
+A basic installation of the Search plugin using the Elasticsearch search engine will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(
+  import('@backstage/plugin-search-backend-module-elasticsearch/alpha'),
+);
+/* highlight-add-end */
+```
+
+#### Search Collators
+
+The following sections outline how you add the Catalog and TechDocs search collators
+
+##### Catalog
+
+A basic installation of the Search plugin with the Catalog collator will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-catalog/alpha'));
+/* highlight-add-end */
+```
+
+##### TechDocs
+
+A basic installation of the Search plugin with the TechDocs collator will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-techdocs/alpha'));
+/* highlight-add-end */
+```
+
+### The Permissions Plugin
+
+A basic installation of the Permission plugin will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-permission-backend/alpha'));
+backend.add(
+  import('@backstage/plugin-permission-backend-module-allow-all-policy'),
+);
+/* highlight-add-end */
+```
+
+> Note: the above includes a default allow all policy, if that is not what you want do not add the 2nd line.
+
+#### Custom Permission Policy
+
+In order to add your own permission policy you'll need to do the following:
+
+```ts
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import { BackstageIdentityResponse } from '@backstage/plugin-auth-node';
+import {
+  PolicyDecision,
+  AuthorizeResult,
+} from '@backstage/plugin-permission-common';
+import {
+  PermissionPolicy,
+  PolicyQuery,
+} from '@backstage/plugin-permission-node';
+import { policyExtensionPoint } from '@backstage/plugin-permission-node/alpha';
+
+class ExampleAllowAllPermissionPolicy implements PermissionPolicy {
+  async handle(
+    _request: PolicyQuery,
+    _user?: BackstageIdentityResponse,
+  ): Promise<PolicyDecision> {
+    return {
+      result: AuthorizeResult.ALLOW,
```

Comment:
and then put a comment in here instead of actual code, something like `// TODO: Add code here that inspects the incoming request and user, and returns AuthorizeResult.ALLOW, AuthorizeResult.DENY, or AuthorizeResult.CONDITIONAL as needed. See the docs at <url> for more information`

Reply:
at that point perhaps the _request and _user can drop their underscores too - it's one of those tricky ones where it's not certain which is more helpful to end users who aren't familiar with ts coding :)